release-notes/Cabal-3.10.1.0.md

```diff
@@ -0,0 +1,159 @@
+
+- Add support for ghc-9.4s -fprof-late flag. [#8528](https://github.com/haskell/cabal/pull/8528)
+
+- This adds the new `late-toplevel` (and its alias `late`) profiling-detail option which enables `-fprof-late`-based automatic cost centre annotations for GHCs that support it ([earliest is GHC 9.4.1](https://downloads.haskell.org/ghc/9.4.1/docs/users_guide/profiling.html#ghc-flag--fprof-late)).
+
+
+### Other changes
+
+- Remove "'-main-is' is not portable" check [#8646](https://github.com/haskell/cabal/issues/8646) [#8651](https://github.com/haskell/cabal/pull/8651)
+
+  'cabal check' no longer complains about '-main-is' flag in 'ghc-options'
+
+- cabal init should not suggest Cabal < 2.0 [#8680](https://github.com/haskell/cabal/issues/8680)
+
+  'cabal init' no longer suggests users to set cabal-version to less than 2.0
+
+- Split up Distribution.Simple.Setup [#8130](https://github.com/haskell/cabal/pull/8130)
+
+  The external interface of 'Distribution.Simple.Setup' has been kept the same, but internally it has been broken up into smaller modules.
+  This improves build times in two ways:
+  1. GHC is superlinear in the size of files, meaning that splitting up a large file can reduce overall compile times.
+  2. Breaking up the module allows dependent modules to refine their imports to just the parts they require, allowing them to start buildling quicker
+  when GHC is run in parrallel make mode ('--ghc-options -j').
+
+- Fix generation of Path_ modules with relocatable [#8219](https://github.com/haskell/cabal/issues/8219) [#8220](https://github.com/haskell/cabal/pull/8220)
+
+  The generation of the functions `minusFileName` and `splitFileName`
+  are now in the same conditional block as their call,
+  preventing generation of inconsistent Paths_ files
+  where those functions are used but not defined.
+
+- Remove warning on option -threaded when building libraries [#774](https://github.com/haskell/cabal/issues/774) [#8431](https://github.com/haskell/cabal/issues/8431) [#8432](https://github.com/haskell/cabal/pull/8432)
+
+  - Accompanied by option `-flink-rts`, option `-threaded` defines the flavour of
+    the ghc RTS library the built library will be linked against. Note that bare
+    ghc does not warn when option `-threaded` is used for building a library
+    either.
+  - Note that the changes require modification of the regression check for issue
+    #774 which can be regarded as a proper test for this PR.
+
+- Add warning about expensive globs [#5311](https://github.com/haskell/cabal/issues/5311) [#8441](https://github.com/haskell/cabal/pull/8441)
+
+  - Now cabal check will emit a warning when package uses
+  recursive globs starting at root of the project
+
+- orders extra stanzas [#8458](https://github.com/haskell/cabal/issues/8458) [#8499](https://github.com/haskell/cabal/pull/8499)
+
+  Ensure that extra-src-dirs, extra sources, and extra other modules all are added using ordNub rather than incidentally alphabetized.
```

Comment:
```suggestion
- Order `extra*` stanzas [#8458](https://github.com/haskell/cabal/issues/8458) [#8499](https://github.com/haskell/cabal/pull/8499)

  Ensure that `extra-src-dirs`, extra sources, and extra other modules all are added using ordNub rather than incidentally alphabetized.
```